pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -318,23 +318,44 @@ func (c *sshClient) GetHardwareDetailsRAM() Output {
 func (c *sshClient) GetHardwareDetailsNics() Output {
 	out := c.runSSH(`cat << 'EOF_VIA_SSH' > nic-info.sh
 #!/bin/sh
-for iname in $(ip a |awk '/state UP/{print $2}' | sed 's/://')
-do
 
-MAC=\""$(ip a | grep -A2 $iname | awk '/link/{print $2}')\""
-SPEED=\""$(ethtool eth0 |grep "Speed:" | awk '{print $2}' | sed 's/[^0-9]//g')\""
-MODEL=\""$( lspci | grep net | head -1 | awk '{$1=$2=$3=""; print $0}' | sed "s/^[ \t]*//")\""
-IP_V4=\""$(ip a | grep -A2 eth0 | sed -n '/\binet\b/p' | awk '{print $2}')\""
-IP_V6=\""$(ip a | grep -A2 eth0 | sed -n '/\binet6\b/p' | awk '{print $2}')\""
+# Iterate over each interface that is "UP"
+for iname in $(ip -o link show | awk -F': ' '{print $2}' | sed 's/@.*//'); do
```

Comment:
@benedikt-bartscher thank you for your PR!

I am looking at this line. Can `ip -o link ...`  be simplified to this?

```
for iname in $(ls /sys/class/net); do
```
At least on my current test system it gets the same list of interfaces.

And I would sort it:

```
for iname in $(ls /sys/class/net | sort); do
```